impl/pom.xml

```diff
@@ -56,6 +60,26 @@
             <groupId>org.yaml</groupId>
             <artifactId>snakeyaml</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.bouncycastle</groupId>
+            <artifactId>bcmail-jdk15on</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.bouncycastle</groupId>
+            <artifactId>bcpkix-jdk15on</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.bouncycastle</groupId>
+            <artifactId>bcprov-jdk15on</artifactId>
```

Comment:
When possible we should use the JCA API, as many of these alg are supported directly in various Java versions.
For example JJWT does this: https://github.com/jwtk/jjwt/blob/b4e5e03152242d867605743ad500af81488dd37b/api/src/main/java/io/jsonwebtoken/lang/RuntimeEnvironment.java#L32

Reply:
ok noted.